### PR TITLE
tests: reduce verbosity of tests

### PR DESCRIPTION
**Pull request by ianvernon**
This PR does a variety of things:

In helpers.bash:
* add function "log", whic prints out name of file and call-stack along with the provided message to log to console.
* add function "get_filename_without_extension", which prints the name of the provided file without its extension.
* add function "redirect_debug_logs" which redirects debug logs (output of set -x) to the provided directory with the filename 'debug.txt'. This allows for debugging of tests if a helper function behaves in an unexpected way without clogging the Console Output in Jenkins.
* 'set +e' in most helper functions, as the helper functions themselves handle errors. Restore this flag if it was set when the helper functions are called using "restore_flag" function at the end of each helper function.
* Do not unset 'x' option in helper functions any more, as we use "redirect_debug_logs" to redirect all debug logs to files.
* use "log" function in various helper functions to provide more information
when tests are run, and replace "echo" in helpers with use of "log"
* declare 'local' variables separately from setting them to the output of a command. Per the Google Bash Style Guide: "Declaration and assignment must be separate statements when the assignment value is provided by a command substitution; as the 'local' builtin does not propagate the exit code from the command substitution."
* add quotation marks around use of various variables in helper functions per recommendations in the Google Bash Style Guide.
* add function "remove_cilium_docker_network" which removes the Cilium Docker network which is created and destroyed in the tests.
* add function "remove_all_containers" which removes all running Docker containers apart from the "cilium-consul" container in the runtime tests.
* add function "test_succeeded" which outputs whether the provided test succeeded in an easy-to-see output to delineate between tests as part of the CI.

In each test:
* Set TEST_NAME variable with "get_filename_without_extension" function.
* Use "redirect_debug_logs" to redirect debug output to log file.
* Replace calls to "echo" with calls to "log" function and add more logs for
easier debugging and descriptions of what is going on during the test in the
CI.
* Add "set -ex" at the start of each test. "set +e" in certain parts of some tests which gracefully handle or expect error cases.
* Use "test_succeeded" function in tests.
* Replace tabs with two spaces in the tests. This is done to reflect the [Bash Style Guide](https://google.github.io/styleguide/shell.xml?showone=Indentation#Indentation) guidelines for indentation.
* Refactor cleanup functions in the tests so that we do not try to gather logs before a test is ran, only after a test completes / exits with failure.
* `tee` the output of each test to a file which is stored with the logs from that test which are picked up in the post-build actions of the Jenkins job. This will allow for easier debugging of test logs as opposed to looking at the Jenkins Console output. 

For runtime tests:
* As we are piping output to 'tee' to achieve the above, run 'set -o pipefail'. If 'set -o pipefail' is not added, even if a test fails, whether it failed will not cause "run-tests" to fail, as 'tee' will always return with exit code 0. Setting 'set -o pipefail' ensures that the script fails if a test fails.

Signed-off by: Ian Vernon <ian@cilium.io>

### CODE REVIEW

**Comment by ianvernon** [comment]
I recommend looking at this PR on a per-commit level, as some of the commits are only whitespace changes.

File: tests/14-policy-enforcement-docker.sh [hunk 1]
```diff
@@ -64,15 +68,15 @@ function check_endpoints_policy_enabled {
...
+  log "checking if ${NUM_EPS} endpoints have policy enforcement disabled"
   cilium endpoint list 
   POLICY_DISABLED_COUNT=`eval ${DISABLED_CMD}`
   if [ "${POLICY_DISABLED_COUNT}" -ne "${NUM_EPS}" ] ; then 
     cilium config
     cilium endpoint list
     abort "Policy Enforcement  should be set to 'Disabled' since policy enforcement was set to never be enabled"
   fi
-  echo  "---- ${NUM_EPS} endpoints have policy enforcement disabled; continuing ----"
+  log  "${NUM_EPS} endpoints have policy enforcement disabled; continuing"
```
Comment (aanm) [inline tests/14-policy-enforcement-docker.sh]: `s/log  "${NUM_EPS}"/log "${NUM_EPS}"`

File: tests/k8s/run-tests.bash [hunk 2]
```diff
@@ -65,10 +65,11 @@ function run_tests(){
     # Set up cilium-lb-ds and cilium-ds
     deploy_cilium ${k8s_version}
 
+
     # Run non IP version specific tests
-    vmssh ${node2} 'set -e; for test in /home/vagrant/go/src/github.com/cilium/cilium/tests/k8s/tests/*.sh; do $test; done'
```
Comment (aanm) [inline tests/k8s/run-tests.bash:69]: What's the reason for this?

**Review by aanm — COMMENTED** [review]
>Replace some of the tests that use echo with log.

Should have a verifier that checks if any 'echo' are being written and tell the developer to use 'log' instead? The same way we are doing with `sleep`

**Comment by tgraf** [comment]
@ianvernon As we merged the new helper changes, this needs a rebase.

File: tests/14-policy-enforcement-docker.sh (hunk 1 above)
Comment (ianvernon) [reply tests/14-policy-enforcement-docker.sh]: What is the difference between what is being substituted vs. what is substituted here? I can't discern a difference.

File: tests/k8s/run-tests.bash (hunk 2 above)
Comment (ianvernon) [reply tests/k8s/run-tests.bash:69]: This allows us to save the output from each test into the folder that is packaged up and saved as an artifact of each build.

File: tests/14-policy-enforcement-docker.sh (hunk 1 above)
Comment (aanm) [reply tests/14-policy-enforcement-docker.sh]: A small space between `log` and `"${NUM_EPS}"` It's a really small nit :)

File: tests/k8s/run-tests.bash (hunk 2 above)
Comment (aanm) [reply tests/k8s/run-tests.bash:69]: Ah got it, it was difficult to read it with that path length

**Review by tgraf — APPROVED** [review]
This is an awesome improvement. Great work!

**Comment by ianvernon** [comment]
Do not merge this yet - for some reason the K8s 1.7 tests aren't getting ran after I rebased. I'm trying to figure this out.

**Comment by aanm** [comment]
@ianvernon I've changed the label to `wip`, feel free to change it back once you've it working

**Comment by ianvernon** [comment]
Please review again as I have added new functionality. Also please check the console output of the most recent build to see if it is to your liking.

**Review by aalemayhu — APPROVED** [review]
IMO this would have been easier to review if it was split into several themed pull requests. 

All of the extra newlines are annoying. Not sure if they were there previously, but overall this seems easier to go through for figuring out which test failed. LGTM.

**Pull request merged**